src/Bindings/ManualBindings_World.cpp

```diff
@@ -950,6 +949,68 @@ static int tolua_cWorld_TryGetHeight(lua_State * tolua_S)
 
 
 
+static int tolua_cWorld_GetHeight(lua_State * tolua_S)
+{
+	// Signature: world:GetHeight(a_World, a_BlockX, a_BlockZ) -> Height
+
+	// Check params:
+	cLuaState L(tolua_S);
+	if (
+		!L.CheckParamSelf("cWorld") ||
+		!L.CheckParamNumber(2, 3) ||
+		!L.CheckParamEnd(4)
+		)
+	{
+		return 0;
+	}
+
+	// Get params:
+	cWorld * self = nullptr;
+	int BlockX = 0;
+	int BlockZ = 0;
+	L.GetStackValues(1, self, BlockX, BlockZ);
+
+	// Call the implementation:
+	L.LogStackTrace();
+	FLOGWARN("cWorld:GetHeight is DEPRECATED, use TryGetHeight instead");
+	auto Height = self->GetHeight(BlockX, BlockZ);
+	L.Push(Height.value_or(0));
+	return 1;
+}
+
+
+
+
+
+static int tolua_cWorld_IsWeatherWetAtXYZ(lua_State * tolua_S)
+{
+	/* Signature: world:IsWeatherWetAtXYZ(Vector) -> boolean */
+
+	// Check params:
+	cLuaState L(tolua_S);
+	if (
+		!L.CheckParamSelf("cWorld") ||
+		!L.CheckParamVector3(2)
+	)
+	{
+		return 0;
+	}
+
+	// Get params:
+	cWorld * self = nullptr;
+	Vector3i BlockPos;
+	L.GetStackValues(1, self, BlockPos);
+
+	// Call the implementation:
+	auto IsWet = self->IsWeatherWetAtXYZ(BlockPos);
+	L.Push(IsWet.value_or(true));
```

Comment:
_not_ raining strikes me as a more useful default, in any case this should be documented in the API docs.

Reply:
It only fails if it's raining in the chunk but the terrain is not loaded so it can't tell where the rain is falling. On the client that would render as rain falling all the way to void i.e. wet everywhere.

Reply:
I could also add a `Try` variant if that's desirable.

Reply:
I guess it doesn't really matter that much, the issue _shouldn't_ come up that often. It's just that everywhere else that IsWeatherWetAtXYZ is used, the default on nullopt is false, rather than true.